app/src/main/java/com/iron/espresso/presentation/sign/SignUpViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package com.iron.espresso.presentation.sign
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+class SignUpViewModel : ViewModel() {
+
+    val signUpEmail = MutableLiveData<String>()
+    val signUpNickname = MutableLiveData<String>()
+    val signUpPassword = MutableLiveData<String>()
+    val isProgressVisible = MutableLiveData<Boolean>()
+
+    private val _checkType = MutableLiveData<CheckType>()
+    val checkType
+        get() = _checkType
+
+    private val _exitIdentifier = MutableLiveData<Boolean>()
+    val exitIdentifier
+        get() = _exitIdentifier
```

Comment:
이렇게쓰면 backing property 사용이 전혀의미가 없어요.
```suggestion
    val exitIdentifier: LiveData<Boolean>
        get() = _exitIdentifier
```
이렇게 타입을 붙여야 외부에서 set이 안됩니다.

Reply:
최대한 activity 에 남기지 않을려고 해보긴 했다만.. 의혹이 든 점이 이렇게 해야되는가에 대해서 고민했거든요..

Reply:
07e5ed1 수정하였습니다.

Reply:
뭐.. 한다면 데이터 바인딩으로 다 할수 있긴하겠지만 너무 그렇게 생각하실 필요는 없을듯

Reply:
So when you don't want your data to be modified use LiveData If you want to modify your data later use MutableLiveData

따라서 데이터를 수정하지 않으려면 LiveData를 사용하십시오. 나중에 데이터를 수정하려면 MutableLiveData를 사용하십시오.

출처 : https://stackoverflow.com/questions/55914752/when-to-use-mutablelivedata-and-livedata#:~:text=Use%20LiveData%20when%20you%20don,values%20by%20calling%20these%20methods.

Reply:
LiveData 라고 붙이지 않으면  _ MutableLiveData 형식으로 만들고나서 get() 하는 방식을 하는 이유가 아무 의미 없어지는건가요??

Reply:
private val _clickIdentifier = MutableLiveData<Boolean>()
val clickIdentifier
        get() = _clickIdentifier

fun onClick(){
     _clickIdentifier.value = true
}

하나의 예를 들어보면
onClick 이라는 함수가 dataBinding 이든 머든 사용이 되면
_clickIdentifier 의 값이 set 되어서 false 였던게 true 로 바뀌고
clickIdentifier 값 또한 get으로 받으니 false-> true 로 바뀔텐데
clickIdentifier : LiveData<Boolean> 을 명시해주지 않으면 요놈도 형태가 MutableLiveData<Boolean>() 테니
set 할수 있을테고.. 그러면 결국 _을 붙인 private 한 변수의 의미가 없게되네요.. 만들필요가 없어지는듯..

결론은 _변수 / 변수 이런식으로 했던 것들에 대해서 _가 붙지않은 변수들의 형태를 모두 LiveData로 해줘야 겠네요..!

@wswon 맞으면 따봉좀 ㅎ

Reply:
665e36d 반영하였습니다.